src/main/Main.js

```diff
@@ -47,75 +50,72 @@ export class Main extends Component {
   }
 
   clickHandlerToggle = (active) => {
-    const activeValue  = this.state[active];
-    
-    activeValue 
-      ? this.setState({ [active]: false })
-      : this.setState({ [active]: true })
-  }
+    const activeValue = this.state[active];
 
-  getGeolocation = () => {
-    if(navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition((position) => {
-        this.setState({
-          latitude: position.coords.latitude,
-          longitude: position.coords.longitude 
-        });                       
-      });
+    if (activeValue) {
+      this.setState({ [active]: false });
     } else {
-        alert("Geolocation API не поддерживается в вашем браузере");
+      this.setState({ [active]: true });
     }
   }
-  
+  getGeolocation = () => {
+    navigator.geolocation.getCurrentPosition((position) => {
+      this.setState({
+        latitude: position.coords.latitude,
+        longitude: position.coords.longitude
+      });
+    });
+  }
+
+  clickHandlerTime = () => {
+    const { showTime } = this.state;
+    this.setState({
+      showTime: !showTime
+    });
+  }
+
   render() {
-    const { 
-      users, 
-      posts, 
-      loadingUsers, 
-      loadingPosts, 
-      latitude, 
+    const {
+      users,
+      posts,
+      loadingUsers,
+      loadingPosts,
+      latitude,
       longitude,
-      activeToggleButton,
-      activeGeolocation 
+      activeGeolocation,
+      showTime
     } = this.state;
 
     return (
-      <main className='main'>
-        <Greeting time={ new Date().getHours() } name='Artem'/>
-        <ToggleButton 
-          clickHandler={ this.clickHandlerToggle }
-          activeText='activeToggleButton'
-          classToggleButton='btn-toggle' 
-          text='show' 
-          toggleText='hide'
-          classToggleComponent='hide'
-          textToggleComponent='Some text'
-          active={ activeToggleButton }
-        />
-        <ToggleButton 
-          clickHandler={ this.clickHandlerToggle }
-          activeText='activeGeolocation'
-          text='Show my geolocation' 
-          toggleText='Hide my geolocation'
-          classToggleComponent='geolocation'
-          textToggleComponent={ `latitude: ${ latitude } longitude: ${ longitude }` }
-          active={ activeGeolocation }
-          text='Show my geolocation' 
-          toggleText='Hide my geolocation'
-          idToggleComponent='geolocation'
+      <main className="main">
+        <Greeting time={new Date().getHours()} name="Artem" />
+        <button onClick={this.clickHandlerTime}>
+          { showTime ? 'Remove time' : 'Show time' }
+        </button>
+        { showTime && <Time /> }
+        <ToggleButton
```

Comment:
geolocation shouldn't be in ToggleButton
geolocation - it is a separate independent component